R/phenotypes.R

```diff
@@ -277,6 +277,177 @@ setPheno = function(pop, h2=NULL, H2=NULL, varE=NULL, corE=NULL,
   return(pop)
 }
 
+#' @title Set population-level phenotypes
+#'
+#' @description
+#' Compute and store population-level phenotypic summaries by applying a
+#' user-supplied function to the individual-level phenotype matrix.
+#' For a \code{MultiPop} object, the function is applied independently to
+#' each population.
+#'
+#' @param pop An object of class \code{\link{Pop-class}},
+#'   \code{\link{HybridPop-class}} or \code{\link{MultiPop-class}}.
+#' @param FUN A function that accepts the individual-phenotype matrix
+#'   (i.e. \code{pop@pheno}) as its first argument and returns 
+#'   a numeric vector of length \code{pop@nTraits} (one value per trait).
+#'   The function will be called as \code{FUN(pop@pheno, ...)}.
+#' @param force Logical. If \code{TRUE}, individual phenotypes will be
+#'   (re)generated by calling \code{\link{setPheno}()} before computing
+#'   the population summary. When \code{force = TRUE} the other phenotype
+#'   generation arguments (see below) are passed to \code{setPheno()}.
+#'   If \code{FALSE}, the function requires an existing \code{pop@pheno}
+#'   matrix and will stop with an error if that matrix is empty.
+#' @param traits Integer vector indicating which trait columns to set.
+#'   If \code{NULL} (the default), all traits are used.
+#' @param h2 Numeric vector of narrow-sense heritabilities (one per trait).
+#'   Passed to \code{\link{setPheno}()} when \code{force = TRUE}.
```

Comment:
```suggestion
#'   Only passed to \code{\link{setPheno}()} when \code{force = TRUE}, otherwise not used.
```

Best to be explicit - I can foresee users asking why they don't get the specifed h2 for population-level phenotype